WordPress/src/main/java/org/wordpress/android/ui/posts/adapters/PostsListAdapter.java

```diff
@@ -378,10 +378,7 @@ public void onItemClick(AdapterView<?> parent, View view, int position, long id)
     private void updatePostUploadProgressBar(ProgressBar view, PostModel post) {
         if (UploadService.isPostUploadingOrQueued(post)) {
             view.setVisibility(View.VISIBLE);
-            // ToDo: update current progress here by some means from the new UploadService
-//            view.setProgress(90);
-            // ToDo: when we have information from PostUploadService, delete the below line
-            view.setIndeterminate(true);
+            view.setProgress(Math.min(90, Math.round(UploadService.getMediaUploadProgressForPost(post) * 100)));
```

Comment:
This is the same value we use [in the visual editor](https://github.com/wordpress-mobile/WordPress-Android/blob/7f91ca46a8585587ed5f17a335a4eef5ac866fcf/libs/editor/WordPressEditor/src/main/assets/ZSSRichTextEditor.js#L1147-L1152) to prevent having the progress bar seemingly stuck at 100% while we're still waiting for a response from the server for media. I actually thought you were referencing this [here](https://github.com/wordpress-mobile/WordPress-Android/pull/6343#discussion-diff-128107931L382) 😉.

I can add an explanatory comment (unless you also have an objection to the approach).